### ISSUE TITLE
Multiple defines with dependencies using UMD style does not work

### ISSUE THREAD

**Issue by remanc** [requirejs/r.js#880]
I'm using r.js to build a project - one of my files has multiple define statements (created using a different project and require). This works for the most part, but there is specifically an issue when a define statement is:
1) Embedded in a UMD style statement
2) Has dependencies

A simple example is this:
```javascript
define('myLib',[],function () {
    return 'myLib';
});

(function (root, factory) {
    // normally would have extra UMD stuff, but this also fails
    define('myLib2',['myLib'], factory);
}(this, function (myLib) {
    return 'myLib2';
}));

define('myLib3',[],function () {
    return 'myLib3';
});
```

In this case, r.js stops parsing after it's done with myLib2, and myLib3 is never added to the list of modules.

Another easy repro is modifying the threeDefines.js file in the build/tests/parse directory - changing:
```javascript
// original line
// define('myLib2',factory);
// new line that breaks:
define('myLib2',[],factory);
```
Will have the same problem, and break the parse.js test.

**Comment by remanc** [comment]
I looked through the code and found a fix that works for my use case and passes other tests as well - does this look correct?
https://github.com/jrburke/r.js/pull/881

Thanks! I also signed the dojo CLA, so my email is on file.

**Comment by jrburke** [comment]
Thanks for the test and first pass at a fix in #881. Unfortunately I did not triage this issue soon enough and the diff in #881 no longer applied. I manually applied the test parts, but on further investigation, a different fix was needed.

The issue was parse.recurse was bailing if any UMD definition was found, instead of checking if the factoryIdentifier value returned was still in the fnExpScope. So that stopped parsing before getting to the third definition. However, making sure to only break out of the processing if factoryIdentifier was still in the function scope, that allows the siblings to be parsed to get to the final define() in the test.

Most recent master snapshot has the change if you want to test it out more:
https://raw.githubusercontent.com/requirejs/r.js/master/dist/r.js
